src/build.rs

```diff
@@ -1114,13 +1114,37 @@ pub fn cbuild(
             let static_libs = if only_cdylib {
                 "".to_string()
             } else {
-                exec.link_line
-                    .lock()
-                    .unwrap()
-                    .values()
-                    .next()
-                    .map_or("", |s| s)
-                    .to_string()
+                if rustc_target.env == "msvc" {
+                    exec.link_line
+                        .lock()
+                        .unwrap()
+                        .values()
+                        .next()
+                        .map_or(String::new(), |s| {
+                            s.trim()
+                                .split(' ')
+                                // Skip forced UCRT library changesº
+                                .filter(|&item| !item.starts_with("/defaultlib"))
+                                .map(|lib| {
+                                    if lib.ends_with(".lib") {
+                                        format!("-l{}", lib.trim_end_matches(".lib"))
+                                    } else {
+                                        lib.to_string()
+                                    }
+                                })
+                                .collect::<Vec<String>>()
+                                .join(" ")
+                        })
+                        .to_string()
+                } else {
+                    exec.link_line
+                        .lock()
+                        .unwrap()
+                        .values()
+                        .next()
+                        .map_or("", |s| s)
+                        .to_string()
+                }
```

Comment:
I think it would be a good idea to move this logic into its own function

```suggestion
            let static_libs = if only_cdylib {
                String::new()
            } else if let Some(libs) = exec.link_line.lock().unwrap().values().next() {
                static_libraries(libs, &rustc_target)?
            } else {
                String::new()
            };
```

in combination with something like

```rust
fn static_libraries(
    link_line: &str,
    rustc_target: &target::Target,
) -> Result<String, std::fmt::Error> {
    use std::fmt::Write;

    let mut buf = String::with_capacity(256);

    for lib in link_line.trim().split(' ') {
        if lib.starts_with("/defaultlib") {
            continue;
        }

        if rustc_target.env == "msvc" && lib.ends_with(".lib") {
            write!(buf, "-l{} ", lib.trim_end_matches(".lib"))?;
        } else {
            write!(buf, "{lib} ")?;
        }
    }

    Ok(buf)
}
```

This function is then testable, so you can add some tests at the bottom of the module to check that this logic does the right thing, independent of the architecture that the code runs on.

Reply:
This is a good idea, I went with the functional approach because with it I can filter out duplicated and empty flags.